eth2/types/src/beacon_state.rs

```diff
@@ -586,13 +586,10 @@ impl<T: EthSpec> BeaconState<T> {
 
     /// Gets the oldest (earliest slot) state root.
     ///
-    /// Spec v0.5.1
+    /// Spec v0.6.3
     pub fn get_oldest_state_root(&self) -> Result<&Hash256, Error> {
-        let lookback = std::cmp::min(
-            self.slot - Slot::from(self.latest_state_roots.len()),
-            self.slot,
-        );
-        let i = self.get_latest_state_roots_index(self.slot - lookback)?;
+        let i = self
+            .get_latest_state_roots_index(self.slot - Slot::from(self.latest_state_roots.len()))?;
```

Comment:
This subtraction will panic in debug mode when `self.slot` is less than 8192 (the length of `latest_state_roots`), and overflow in a "safe" way that's caught by `get_latest_state_roots_index` in release mode.

I think we could use `Slot::from(self.slot.as_u64().saturating_sub(self.latest_state_roots.len()))`, that way we get the 0th element of the ring buffer if it hasn't yet wrapped around, or otherwise the same as above.

Reply:
Never mind the above! `Slot` subtraction is already saturating, as you point out :)